rclcpp/include/rclcpp/clock.hpp

```diff
@@ -137,6 +137,45 @@ class Clock
     Duration rel_time,
     Context::SharedPtr context = contexts::get_global_default_context());
 
+  /**
+   * Check if the clock is valid.
+   *
+   * An invalid clock is either a clock that is RCL_CLOCK_UNINITIALIZED or a clock with zero-time.
+   * In other words, an uninitialized clock.
+   *
+   * \return true if clock was or became valid
+   */
+  RCLCPP_PUBLIC
+  bool
+  is_valid();
+
+  /**
+   * Wait for clock to become valid.
+   *
+   * \param context the context to wait in.
+   * \return true if clock was or became valid
+   */
+  RCLCPP_PUBLIC
+  bool
+  wait_for_valid(Context::SharedPtr context = contexts::get_global_default_context());
```

Comment:
this throws exception with current code, if so probably doc needs to be described accordingly here.

Reply:
https://github.com/ros2/rclcpp/pull/2040/commits/a0091d0e5527fa45a09a77c70c98b8be82fe5b98